### ISSUE TITLE
CheckNTP UNKNOWN: NTP command Failed

### ISSUE THREAD

**Issue by dzeleski** [sensu-plugins/sensu-plugins-ntp#12]
This just occured site wide for us. I have no changes to our code base and Im not sure whats going on here:

```
/opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-ntp.rb -w 700 -c 1000
CheckNTP UNKNOWN: NTP command Failed
```
```
 ntpq -c "rv 0"
associd=0 status=4615 leap_add_sec, sync_ntp, 1 event, clock_sync,
version="ntpd 4.2.6p5@1.2349-o Wed Jan 20 15:22:38 UTC 2016 (1)",
processor="x86_64", system="Linux/3.10.0-327.13.1.el7.x86_64", leap=01,
stratum=4, precision=-24, rootdelay=41.747, rootdisp=48.012,
refid=10.192.1.31,
reftime=dc1182bc.0491f3dc  Fri, Dec 30 2016 20:05:00.017,
clock=dc118408.6e8df8a6  Fri, Dec 30 2016 20:10:32.431, peer=21279, tc=6,
mintc=3, offset=0.310, frequency=17.705, sys_jitter=0.000,
clk_jitter=0.298, clk_wander=0.007
```
```
 ntpq -c "rv 0 stratum,offset"
stratum=4, offset=0.937
```

I noticed that status is "4615 leap_add_sec" is that due to it being a leap second month? If that is the case the check shouldnt return with command failed I would think?

**Comment by kevinschmidt** [comment]
the issue is this line of code:

`source_field_status = config[:unsynced_status] == 'ok' ? 6 : /status=[0-3]([0-9])[0-9a-f]{2}/.match(`ntpq -c "rv 0"`)[1].to_i`

as leap_add_sec is 4615 which starts with a 4, [0-3] does not match and the whole command fails.

**Comment by dzeleski** [comment]
Apparently I cant read code properly after being off for a week... TY that makes sense. Is that something that is safe to fix without causing other issues? NTP isnt my strong point.

**Comment by kevinschmidt** [comment]
I'm no expert but `source_field_status` does not use the first digit and 4615 is a valid code. How the status hex is put together is a bit complex, I don't think I understand it properly but not all values fit the regex above.

**Comment by nullify005** [comment]
I've just temporarily made it go away by changing the `status=[0-3]` blah to `status=[0-4]` blah
I'm sure that there's a better fix than this as @kevinschmidt indicates

**Comment by crscybraics** [comment]
Did https://github.com/sensu-plugins/sensu-plugins-ntp/commit/57ca6bafc06c4460f592df049b8eb5a042af9876 make it into 1.0.0 version of the gem?  That commit contains the code fix but I'm resorting to fixing locally.

**Comment by rwdlander** [comment]
Did commit https://github.com/sensu-plugins/sensu-plugins-ntp/commit/57ca6bafc06c4460f592df049b8eb5a042af9876 make it into 1.0.0 of the gem?  That commit contains the fix, but I'm having to fix locally even after re-downloading the gem through sensui-install.

**Comment by dzeleski** [comment]
@rwdlander I dont see it: 

https://github.com/sensu-plugins/sensu-plugins-ntp/commit/3ac74d45e6b9c60d8c3dadb6ae4f9d50ed2b1f48 

https://github.com/sensu-plugins/sensu-plugins-ntp/blob/3ac74d45e6b9c60d8c3dadb6ae4f9d50ed2b1f48/bin/check-ntp.rb#L65

**Comment by eheydrick** [comment]
I just released [1.0.1](https://rubygems.org/gems/sensu-plugins-ntp/versions/1.0.1) which includes the fix for the leap status that @rwdlander mentioned.
